### PR TITLE
DEX-332 percentage completion calculation

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- percentage completion calculation reported as discussed
- Updated a few tests to validate it

---

### CODE REVIEW

File: app/modules/asset_groups/models.py [hunk 1]
```diff
@@ -247,6 +247,39 @@ def commit(self):
...
+                size_range = (
+                    stage_base_sizes[AssetGroupSightingStage.curation]
+                    - stage_base_sizes[self.stage]
+                )
+                complete_jobs = [job for job in self.jobs if not job['active']]
+                completion += size_range * (len(complete_jobs) / len(self.jobs))
+        elif self.stage == AssetGroupSightingStage.processed:
+            assert len(self.asset_group_sightings) == 1
+            size_range = 100 - stage_base_sizes[self.stage]
+            sighting_completion = self.asset_group_sightings[0].get_completion()
```
Comment (hwindsor) [inline app/modules/asset_groups/models.py:277]: I am thinking that maybe asset_group_sightings is the wrong name. This is the sighting object associated with the asset_group_sighting. 
I could find a way to create a 1:1 sqlalchemy mapping so it has to be a 1:many and in this case, many = 1, hence the assert

File: app/modules/asset_groups/schemas.py [hunk 2]
```diff
@@ -27,11 +27,18 @@ class DetailedAssetGroupSightingSchema(BaseAssetGroupSightingSchema):
     Detailed Asset_group_sighting schema exposes all useful fields.
     """
 
+    completion = base_fields.Function(lambda obj: obj.get_completion())
```
Comment (hwindsor) [inline app/modules/asset_groups/schemas.py:30]: @karenc I found this example online
https://marshmallow.readthedocs.io/en/stable/examples.html
the magic of how this lambda allows you to get at the object to call the method is a bit of a mystery 
I have no idea if this is a really clever idea or a bad hack :)
Any comments?

File: app/modules/asset_groups/schemas.py [hunk 3]
```diff
@@ -27,11 +27,15 @@ class DetailedAssetGroupSightingSchema(BaseAssetGroupSightingSchema):
     Detailed Asset_group_sighting schema exposes all useful fields.
     """
 
+    completion = base_fields.Function(AssetGroupSighting.get_completion)
```
Comment (karenc) [inline app/modules/asset_groups/schemas.py:30]: > the magic of how this lambda allows you to get at the object to call the method is a bit of a mystery
> I have no idea if this is a really clever idea or a bad hack :)
> Any comments?

@hwindsor The difference between `AssetGroupSighting.get_compeltion` and `lambda obj: obj.get_completion()` is:

```
AssetGroupSighting.get_completion(asset_group_sighting)
```

and

```
asset_group_sighting.get_completion()
```

I think `lambda obj: obj.get_completion()` is probably better...?

`lambda obj: obj.get_completion()` is basically:

```
def f(obj):
    return obj.get_completion()
```

It's not a clever idea or bad hack, it's just a normal way of creating a callable

**Review by colinwkingen — APPROVED** [review]
LGTM. We can adjust things or get fancier post MVP if we decide it's needed.

**Pull request merged**